miio/dreamevacuum_miot.py

```diff
@@ -0,0 +1,376 @@
+
+    @property
+    def timer_enable(self) -> str:
+        return self.data["timer_enable"]
+
+    @property
+    def start_time(self) -> str:
+        return self.data["start_time"]
+
+    @property
+    def stop_time(self) -> str:
+        return self.data["stop_time"]
+
+    @property
+    def map_view(self) -> str:
+        return self.data["map_view"]
+
+    @property
+    def volume(self) -> str:
+        return self.data["volume"]
+
+    @property
+    def voice_package(self) -> str:
+        return self.data["voice_package"]
+
+
+class DreameVacuumMiot(MiotDevice):
+    """Interface for Vacuum 1C STYTJ01ZHM (dreame.vacuum.mc1808)"""
+
+    def __init__(
+        self, ip: str, token: str = None, start_id: int = 0, debug: int = 0
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug)
+
+    @command(
+        default_output=format_output(
+            "\n",
+            "Battery level: {result.battery_level}\n"
+            "Brush life level: {result.brush_life_level}\n"
+            "Brush left time: {result.brush_left_time}\n"
+            "Charging state: {result.charging_state.name}\n"
+            "Cleaning mode: {result.cleaning_mode.name}\n"
+            "Device fault: {result.device_fault.name}\n"
+            "Device status: {result.device_status.name}\n"
+            "Filter left level: {result.filter_left_time}\n"
+            "Filter life level: {result.filter_life_level}\n"
+            "Life brush main: {result.life_brush_main}\n"
+            "Life brush side: {result.life_brush_side}\n"
+            "Life sieve: {result.life_sieve}\n"
+            "Map view: {result.map_view}\n"
+            "Operating mode: {result.operating_mode.name}\n"
+            "Side cleaning brush left time: {result.brush_left_time2}\n"
+            "Side cleaning brush life level: {result.brush_life_level2}\n"
+            "Timer enabled: {result.timer_enable}\n"
+            "Timer start time: {result.start_time}\n"
+            "Timer stop time: {result.stop_time}\n"
+            "Voice package: {result.voice_package}\n"
+            "Volume: {result.volume}\n",
+        )
+    )
+    def status(self) -> DreameVacuumStatus:
+        """State of the vacuum."""
+
+        return DreameVacuumStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    def send_action(self, siid, aiid, params=None):
+        """Send action to device."""
+
+        # {"did":"<mydeviceID>","siid":18,"aiid":1,"in":[{"piid":1,"value":2}]
+        if params is None:
+            params = []
+        payload = {
+            "did": f"call-{siid}-{aiid}",
+            "siid": siid,
+            "aiid": aiid,
+            "in": params,
+        }
+        return self.send("action", payload)
+
+    # siid 3: (Robot Cleaner): 2 props, 2 actions
+    # aiid 1 Start Sweep: in: [] -> out: []
```

Comment:
Please remove these commentaries, they were simply there as a development hint in the generated code, but shouldn't be there for the finalized one.